src/Casbin.AspNetCore/ServiceCollectionExtension.cs

```diff
@@ -1,13 +1,49 @@
-﻿using Microsoft.Extensions.DependencyInjection;
+﻿using System;
+using Casbin.AspNetCore.Abstractions;
+using Casbin.AspNetCore.Policy;
+using Casbin.AspNetCore.Transformers;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using NetCasbin;
+using NetCasbin.Model;
 
 namespace Casbin.AspNetCore
 {
     public static class ServiceCollectionExtension
     {
-        public static IServiceCollection AddCasbinAuthorization(this IServiceCollection services)
+        public static IServiceCollection AddCasbinAuthorizationCore(this IServiceCollection services, Action<CasbinAuthorizationCoreOptions> configure)
         {
-
+            var options = CasbinAuthorizationCoreOptions.GetDefault();
+            configure(options);
+            CheckOptions(options);
+            services.TryAdd(ServiceDescriptor.Describe(typeof(Model),
+                p => options.ModelFactory?.Invoke(),
```

Comment:
Is this `ModelFactory` necessary? I didn't see it's usage, a model shouldn't change often after the server was up

Reply:
All the policies are in the model, We can load policies here, I think the model lifetime should be singleton at some usage now.